app/Observers/UserObserver.php

```diff
@@ -22,8 +25,19 @@ public function created(User $user)
             ->where('name', '=' , 'welcome-message')
             ->firstOrFail();
 
-        if (!$notificationTemplate->disabled) {
-            $user->notify($notificationTemplate->getDynamicNotification(compact('user')));
+        if (!$notificationTemplate->disabled && app(MailSettings::class)->mail_password !== null) {
```

Comment:
like i mentioned earlier, i would like the MailSettings be added to the constructor instead of calling app directly.